### DISCUSSION TITLE
Problem with implementing the tiptap editor

### DISCUSSION THREAD

**Discussion by Sch-Tim** [alpinejs/alpine#3021] [7. Bugs (archived)]
Hello, 
I'm trying to use the tiptap editor with Alpine v3 (and Livewire). Unfortunately I'm having the same struggle as the folks [here](https://github.com/ueberdosis/tiptap/issues/1515). While there are some ideas on how to solve this, I couldn't get it to work. 

Since the discussion is now more than a year old, I want to try it here. 

How can I get tiptap to work with alpine v3? Unfortunately downgrading is not an option since I am using some features of v3. 

One more info: I need to have multiple instances of the editor.

[Codesandbox](https://codesandbox.io/s/pensive-river-bkxqoc?file=/index.html)

Thanks in advance 

Tim

**Comment by samwillis** [comment]
Although I haven't touched TipTap in a year, I think I may have some knowledge that can help on this as I contributed the Vue3 support to TipTap and recognise the exact errors there!

Alpine.js v3 uses the ["reactive" engine from Vue.js V3](https://alpinejs.dev/advanced/reactivity), which uses JS Proxies for a bunch of clever stuff, but is "incompatible" with ProseMirror/TipTap. The trick with Vue was to use a [ShallowRef()](https://vuejs.org/api/reactivity-advanced.html#shallowref) to hold the Editor object, this stops the reactive engine from trying to track all properties and methods within TipTap. (we then did a bunch of stuff to make the toolbars reactive but my memory gives up at that point)

I don't know how to access the same `ShallowRef()` functionality with Alpine... will probably going looking now...

**Reply by Sch-Tim** [reply]
That would be great. Thanks to your idea with the reactivity and the need to find a way to suppress it, I found this: https://sebastiandedeyne.com/non-reactive-data-in-alpine-js/

I will look into it and report.

**Reply by earthboundkid** [reply]
Sorry to revive a dead conversation, I just wanted to say thank you for writing that up because I'm working with Vimeo's player objects, and apparently it also needs to be stored as non-reactive data, and you saved me from having to figure out a good method myself.
